components/SearchResultDetails.vue

```diff
@@ -169,43 +206,82 @@ const localeStore = useLocaleStore()
 const specialtiesStore = useSpecialtiesStore()
 
 // Form Data
+const activeProfessional = computed(() => resultsStore.activeProfessional)
+const activeFacility = computed(() => resultsStore.activeFacility)
 
 const healthcareProfessionalName = computed(() => {
     const name = formatHealthcareProfessionalName(
-        resultsStore.activeResult?.professional.names,
+        activeProfessional.value?.names,
         localeStore.activeLocale.code as Locale
     )
     return name
 })
 
-const healthcareProfessionalDegrees = computed(() => {
-    const healthcareProfessionalDegreesText
-        = resultsStore.activeResult?.professional.degrees.join(', ')
-    return healthcareProfessionalDegreesText
-})
+const healthcareProfessionalDegrees = computed(() => activeProfessional.value?.degrees?.join(', '))
+
 const specialties = computed(() => {
-    const specialties
-        = resultsStore.activeResult?.professional.specialties
-
-    const specialtiesDisplayText = specialties?.map(specialty => {
-        const specialtyDisplayText
-            = specialtiesStore.specialtyDisplayOptions.find(
-                options => options.code === specialty
-            )?.displayText
-        return specialtyDisplayText
-    })
+    const specialtiesList = activeProfessional.value?.specialties
+
+    if (!specialtiesList || specialtiesList.length === 0) return []
+
+    const specialtiesDisplayText = specialtiesList
+        .map(specialty => specialtiesStore.specialtyDisplayOptions.find(
+            options => options.code === specialty
+        )?.displayText)
+        .filter((text): text is string => !!text)
 
     return specialtiesDisplayText
 })
+
 const facilityName = computed(() => {
-    const englishName = resultsStore.activeResult?.facilities[0].nameEn
-    const japaneseName = resultsStore.activeResult?.facilities[0].nameJa
+    const facility = activeFacility.value
+    if (!facility) return ''
+
+    const englishName = facility.nameEn
+    const japaneseName = facility.nameJa
     return localeStore.activeLocale.code === Locale.JaJp ? japaneseName : englishName
 })
 
+type MappedProfessional = {
+    id: string
+    displayName: string
+    degrees: string[]
+}
+
+const facilityProfessionals = computed(() => {
+    const allProfessionals = activeFacility.value?.healthcareProfessionals
+    const currentProfessionalId = activeProfessional.value?.id
+
+    if (!allProfessionals || allProfessionals.length === 0) return []
+
+    const mapProfessional = (hp: HealthcareProfessional): MappedProfessional => ({
```

Comment:
curious, was this AI written?  don't see many people naturally writing nested functions. 
Not bad, just curious